### PR TITLE
Support window aggregation in batch

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/Traverser.java [hunk 1]
```diff
@@ -172,4 +172,30 @@ public T next() {
     static <T> Traverser<T> over(T... items) {
         return Traversers.traverseArray(items);
     }
+
+    /**
+     * Returns a traverser traversing over the longest prefix of elements taken
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/Traverser.java]: 1. Just "traverser over" would be enough.
2. There is no clear treatment of `null` returned from the wrapped traverser. `null` shouldn't be passed into the predicate, we should treat it as if the predicate returned `false`. We must also document that `takeWhile` handles only`null`-terminated sequences.

Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/Traverser.java]: Thanks, I completely overlooked it. I also added test for it.

**Comment by mtopolnik** [comment]
verify

**Comment by mtopolnik** [comment]
verify

**Comment by mtopolnik** [comment]
verify

**Pull request merged**